webfrontend/src/components/popup/request-tag-modal.tsx

```diff
@@ -0,0 +1,227 @@
+  const imageInput = useRef<HTMLInputElement>(null);
+  const [description, setDescription] = useState("");
+  const [attach, setAttach] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tagRequestMessageContent, setTagRequestMessageContent] =
+    useState<string>("");
+  const [show, setShow] = useState(false);
+  const [showMessage, setShowMessage] = useState(false);
+
+  const handleAddImage = useCallback((event: React.ChangeEvent<any>) => {
+    setImagePreview(URL.createObjectURL(event.target.files[0]));
+    setImageToBeSubmitted(event.target.files[0]);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    setShow(false);
+    setErrMessage("");
+  }, []);
+
+  const handleShow = useCallback(() => {
+    setShow(true);
+    setErrMessage("");
+  }, []);
+
+  const submitForm = useCallback(
+    (event: React.SyntheticEvent<EventTarget>) => {
+      event.preventDefault();
+      if (tag === "") {
+        setErrMessage("Tag cannot be empty");
+        tagInput.current?.classList.add("is-invalid");
+        return;
+      } else {
+        tagInput.current?.classList.remove("is-invalid");
+      }
+      setErrMessage("");
+      const requestBody: { [k: string]: any } = {
+        tag: tag,
+        description: description,
+        attach: attach,
+      };
+      if (imageToBeSubmitted !== "") {
+        requestBody.img = imageToBeSubmitted;
+      }
+      if (!isLoading) {
+        setIsLoading(true);
+        fetch("/add_tag_request", postRequestContent(requestBody)).then(
+          async (response) => {
+            setIsLoading(false);
+            if (response.status !== 200) {
+              triggerErrorMessage();
+            } else {
+              setTag("");
+              setDescription("");
+              setAttach(false);
+              setImagePreview("");
+              const text = await response.text();
+              if (text === "tag already present/requested") {
+                setTagRequestMessageContent("Tag is already present/requested");
+              } else {
+                setTagRequestMessageContent(
+                  "Your request is sent, our admin will review your request."
+                );
+              }
+              setShowMessage(true);
+            }
+          }
+        );
+      }
+    },
+    [attach, description, imageToBeSubmitted, isLoading, tag]
+  );
+
+  const handleCloseMessage = useCallback(() => {
+    setShowMessage(false);
+    setTagRequestMessageContent("");
+    if (
+      tagRequestMessageContent ===
+      "Your request is sent, our admin will review your request."
+    ) {
+      handleClose();
+    }
+  }, [handleClose, tagRequestMessageContent, setTagRequestMessageContent]);
+
+  return (
+    <>
+      <img src={imgLink} alt={"request-tag"} onClick={handleShow} />
+
+      <Modal className="request-tag-modal" show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
```

Comment:
Ideally, the component should only contain the modal, instead of a modal and the image.